### DISCUSSION TITLE
How does the client actually enforce the required properties on table items?

### DISCUSSION THREAD

**Discussion by oliverw** [Azure/azure-mobile-apps#591] [Q&A]
Looking at the code there seems to be no interface contract for the properties provided by the `DatasyncClientData` base class. Why is that?

**Comment by adrianhall** [comment]
Do you have a use-case where they would be required?  I didn't think one was required.

**Reply by oliverw** [reply]
No, use case. Its just something that you rarely experience with Microsoft SDKs. Replacing `DatasyncClientData` with something like the base class below kind felt like rigging a time bomb 😄 

```
    public class ReactiveDatasyncClientData : ReactiveObject
    {
        [JsonProperty("id")]
        public string Id { get; set; }

        [JsonProperty("deleted")]
        public bool Deleted { get; set; }
        
        [JsonProperty("updatedAt")]
        public DateTimeOffset UpdatedAt { get; set; }
        
        [JsonProperty("version")]
        public string Version { get; set; }
    }

```

**Reply by adrianhall** [reply]
The DatasyncClientData is just a convenience - it isn't actually required, and you can generate your own rather easily.
